nmostesting/suites/IS0502Test.py

```diff
@@ -157,13 +162,25 @@ def activate_check_version(self, resource_type, resource_list):
                         valid, response = self.do_request("GET", self.node_url + resource_type + "/" + is05_resource)
                         if not valid:
                             return False, "Node API did not respond as expected: {}".format(response)
+                        new_is04_resource = response.json()
 
-                        new_ver = response.json()["version"]
+                        new_ver = new_is04_resource["version"]
 
                         if self.is05_utils.compare_resource_version(new_ver, current_ver) != 1:
                             return False, "IS-04 resource version did not change when {} {} was activated" \
                                           .format(resource_type.rstrip("s").capitalize(), is05_resource)
 
+                        if resource_type == "receivers" and current_caps_ver:
+                            # the 'caps' version shouldn't change unless something else in 'caps' has changed
+                            # and that shouldn't happen as a result of the activation
+                            new_caps = new_is04_resource["caps"]
+                            new_caps_ver = new_caps["version"]
+                            if self.is05_utils.compare_resource_version(new_caps_ver, current_caps_ver) != 0:
+                                new_caps["version"] = current_caps_ver
+                                if compare_json(new_caps, current_caps):
+                                    return False, "IS-04 caps version changed when {} {} was activated" \
```

Comment:
In other areas, there's links to the spec. Could that be useful here?

Reply:
Yes.
But.

Test case functions themselves return `TestResult` objects which are constructed by the  functions `FAIL`, `WARNING`, etc. of the `Test` object that is passed into every test case function. (The `Test` object is constructed by the framework with a `name` and `description` via introspection of the test case function.)

We've ended up with two styles of utility function in the test suites:
* simple 'check' functions return a Boolean pass/fail indicator and a message
* 'do_test' functions which either return a `TestResult`, or raise an `NMOSTestException` (which has an embedded `TestResult`) when there's an error - these need to have the `Test` object passed in from the test case function

In order to be able to pass back the 'More Information' link as well as the message and the fail/pass-or-warning flag, we'll have to refactor `activate_check_version` and friends into the `do_test` style.

Reply:
Switching `activate_check_version(self, ...)` to `do_test_activate_version(self, test, ...)` is quite straightforward and this particular `return` statement would become:
```python
                                    api = self.apis[RECEIVER_CAPS_KEY]
                                    return test.FAIL("IS-04 caps version changed when {} {} was activated"
                                                     .format(resource_type.rstrip("s").capitalize(), is05_resource),
                                                     "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
                                                     "/docs/1.0._Receiver_Capabilities.html#capabilities-version"
                                                     .format(api["spec_branch"]))
```

The similar functions `activate_check_parked` and `activate_check_subscribed` are used in sequence in several test case functions, and through the codebase we don't tend to check `TestResult` objects. All three functions could be switched over to just return None on success and `raise NMOSTestException` on failure. That makes "returning" warnings problematic, but these functions don't do that currently anyway...